### ISSUE TITLE
Mobilenet Pretrained model for 4  class custom dataset displays/predicts always 1 (same) class name for any input image

### ISSUE THREAD

**Issue by Anishauday** [keras-team/keras#14303]
I will try to elaborate my prob. Currently I'm trying transfer learning using a pretrained mobilenet model that was initally trained for 1000 classes. My custom data has just 4 classes.

train_batches = ImageDataGenerator(preprocessing_function=tf.keras.applications.mobilenet.preprocess_input).flow_from_directory(directory=train_path, target_size=(224,224), batch_size=10)
valid_batches = ImageDataGenerator(preprocessing_function=tf.keras.applications.mobilenet.preprocess_input).flow_from_directory(directory=valid_path, target_size=(224,224), batch_size=10)
test_batches = ImageDataGenerator(preprocessing_function=tf.keras.applications.mobilenet.preprocess_input).flow_from_directory(directory=test_path, target_size=(224,224), batch_size=10, shuffle=False) horizontal_flip = True) #for randomly flipping half of the image

Initial model summary:
.....
......
**conv_pw_13_relu (ReLU) (None, 7, 7, 1024) 0

global_average_pooling2d (Gl (None, 1024) 0

reshape_1 (Reshape) (None, 1, 1, 1024) 0

dropout (Dropout) (None, 1, 1, 1024) 0

conv_preds (Conv2D) (None, 1, 1, 1000) 1025000

reshape_2 (Reshape) (None, 1000) 0

predictions (Activation) (None, 1000) 0
Total params: 4,253,864
Trainable params: 4,231,976
Non-trainable params: 21,888**

Modified Model summ
.......
**conv_dw_13_bn (BatchNormaliz (None, 7, 7, 1024) 4096

conv_dw_13_relu (ReLU) (None, 7, 7, 1024) 0

conv_pw_13 (Conv2D) (None, 7, 7, 1024) 1048576

conv_pw_13_bn (BatchNormaliz (None, 7, 7, 1024) 4096

conv_pw_13_relu (ReLU) (None, 7, 7, 1024) 0

global_average_pooling2d (Gl (None, 1024) 0**

dense (Dense) (None, 4) 4100
Total params: 3,232,964
Trainable params: 1,867,780
Non-trainable params: 1,365,184

#Train the model
model.compile(optimizer=Adam(lr=0.0001), loss='categorical_crossentropy', metrics=['accuracy'])

**Epoch 1/5
227/227 - 136s - loss: 0.0492 - accuracy: 0.9859 - val_loss: 0.1093 - val_accuracy: 0.9600
Epoch 2/5
227/227 - 134s - loss: 0.0030 - accuracy: 0.9996 - val_loss: 0.1655 - val_accuracy: 0.9600
Epoch 3/5
227/227 - 142s - loss: 0.0026 - accuracy: 1.0000 - val_loss: 0.1175 - val_accuracy: 0.9600
Epoch 4/5
227/227 - 139s - loss: 0.0015 - accuracy: 0.9996 - val_loss: 0.1045 - val_accuracy: 0.9600
Epoch 5/5
227/227 - 140s - loss: 6.2960e-04 - accuracy: 1.0000 - val_loss: 0.1643 - val_accuracy: 0.9600**

label_names = sorted(train_batches.class_indices.items(), key=lambda pair:pair[1])
label_names = np.array([key.title() for key, value in label_names])
label_names
**Output**: array(['Class 1', 'Class 2', 'Class 3', 'Class 4'], dtype='<U7')

img_path = image.load_img('./test/Class 2/img.jpg',target_size=(224,224)) 
test_image = image.img_to_array(img_path)          
test_image = np.expand_dims(test_image, axis=0) 
test_image = np.array(test_image)/127.5-1
plt.imshow(test_image.squeeze())

result = model.predict(test_image)              
result1 = result[0]                                 
for i in range (4):                                
    if result1[i] == 1.:                          
        break;
p=model.predict(test_image)
print(p.shape)
predict_class = label_names[np.argmax(p[0][0], axis=-1)]
plt.imshow(test_image.squeeze())
plt.title(predict_class)
plt.show()

**Problem**: While predicting, the model shows always one class name(Class 1) for any type image input. I'm clueless as to how to fix this.. I assume mistake should be with preprocessing the test image.

Thanks in advance. Pease help

**Comment by itsshaikaslam** [comment]
i think while training you are using mobilenet_preprocess_input function but while inference you are not using the same preprocessing function thats where every image got classified as one class ....

to fix this  you can use below code

pred_dir  is the place of folder where your images present 

def prepare_image(file):
    img = image.load_img(file, target_size=(224, 224))
    img_array = image.img_to_array(img)
    img_array_expanded_dims = np.expand_dims(img_array, axis=0)
    return tf.keras.applications.mobilenet.preprocess_input(img_array_expanded_dims)
  
  


#modelmobile1  is your trained model 

for item in os.listdir(pred_dir):
  imagepath = os.path.join(pred_dir,item)
  preprocessed_image = prepare_image(imagepath)
  predictions = modelmobile1.predict(preprocessed_image)
  result.append([item,np.argmax(predictions,axis=1)])


Hope this helps you....
